### PR TITLE
Add extension bleed SSL KB

### PR DESCRIPTION
**Pull request by Zakaria-Latif**
Add extension bleed SSL KB

### CODE REVIEW

File: WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/description.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+The application is detected to be vulnerable to SSL Extension Bleed, a type of vulnerability that exploits improper handling of SSL/TLS extensions during the handshake process.
```
Comment (elyousfi5) [inline WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/description.md:1]: This looks like a start of a technical detail. A description should simply define what is the vulnerability.

File: WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/meta.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "risk_rating": "potentially",
+  "short_description": "Vulnerability in SSL/TLS extension handling can lead to sensitive data exposure.",
+  "references": {
+    "CWE-200: Data Exposure": "https://heartbleed.com/",
```
Comment (PiranhaSa) [inline WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/meta.json:5]: Why you have the CWE as a title for website that explain the whole vulnerability? I think you can change the title and add another link for the CWE : https://cwe.mitre.org/data/definitions/200.html

**Review by ybadaoui-ostorlab — APPROVED** [review]

File: WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/recommendation.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    
+    - **Validation:** After upgrading, test your application to ensure compatibility and functionality with the new library versions.
+
+- **Implement Strong Cipher Suites:** 
+  
+  - **Action:** Configure your server to prioritize strong cipher suites, such as those based on AES-GCM or ChaCha20, and disable weak ones (e.g., RC4, DES, 3DES).
+
+  - **Best Practice:** Use tools like SSL Labs' SSL Test to evaluate your server's SSL/TLS configuration and receive recommendations for improving security.
+
+  - **Validation:** Regularly review the configuration of your web servers, load balancers, and any proxies handling SSL/TLS to ensure compliance with best practices.
```
Comment (3asm) [inline WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/recommendation.md:17]: I really don't like this format action best practice, validation. Is simpler recommendation not possible?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/description.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+SSL Extension Bleed is a vulnerability that exploits improper handling of SSL/TLS extensions during the handshake process. It can lead to the exposure of sensitive information, such as plaintext data from previous connections, session keys, or even private keys.
+
+This vulnerability allows attackers to:
+
+- **Extract Sensitive Information:** Attackers can access data stored in memory, potentially revealing SSL session keys or plaintext data transmitted over earlier SSL connections.
+
+- **Hijack Sessions:** If session keys are disclosed, attackers may gain unauthorized access to existing SSL sessions, allowing them to access user accounts or sensitive information without permission.
```
Comment (ohachimOs) [inline WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/description.md:7]: We need an example

File: WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/description.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+**Example Scenario:**
+
+Imagine a web server that supports SSL/TLS and allows a specific extension, such as `SessionTicket TLS`, to optimize session resumption during the handshake process. However, due to improper memory management or a flaw in the handling of these extensions, some sensitive data from previous SSL sessions (such as session keys or portions of plaintext data) remains in memory.
+
+An attacker who can repeatedly initiate SSL/TLS handshakes may craft malicious SSL extensions or probe the server in a way that causes the server to unintentionally leak data from memory during the handshake. As a result, the attacker might recover session keys used in earlier connections, which could then be used to decrypt previously encrypted communication or hijack active user sessions.
```
Comment (nmasdoufi-ol) [inline WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/description.md:13]:
```suggestion
An attacker who can repeatedly initiate SSL/TLS handshakes may craft malicious SSL extensions or probe the server to unintentionally leak data from memory during the handshake. As a result, the attacker might recover session keys used in earlier connections, which could then be used to decrypt previously encrypted communication or hijack active user sessions.
```

Comment (deadly-panda) [inline WEB_SERVICE/WEB/_POTENTIALLY/SSL_EXTENSION_BLEED/description.md:13]: I would  expect more details on how these malicious SSL extensions or how the server can unintentionally leak data..

**Review by PiranhaSa — APPROVED** [review]

**Review by elyousfi5 — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**